src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -184,7 +187,13 @@ private static void setModules(TestInputConfiguration.Builder testInputConfigBui
                     new ModuleInputConfiguration.Builder();
             setModuleName(moduleInputConfigBuilder, inputFilePath, lines.get(lineNo));
             setProperties(moduleInputConfigBuilder, inputFilePath, lines, lineNo + 1);
-            testInputConfigBuilder.addChildModule(moduleInputConfigBuilder.build());
+            if (TREE_WALKER_MODULE_NAME.equals(lines.get(lineNo))) {
+                testInputConfigBuilder.setTreeWalkerModuleInputConfiguration(
```

Comment:
Please hide all logic of special handling of Treewalker at testInputConfigBuilder

Treewalker is just another module. As it appears we add it to configbuilder. And config builder internally figure out how to replace all properties new Treewalker to existing Treewalker created by default.

Reply:
Do you mean I remove this condition of checking is the module TreeWalker or not ?

and add it as child module ? just like another modules

Reply:
You will need this condition in Builder class. Yes, almost like just another module.

Reply:
Got it

Reply:
Is it ok ?